osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ConfigurationService.java

```diff
@@ -672,12 +672,19 @@ public void generateAndReplaceKeys(final MessageMetadata messageMetadata)
     final SmartMeter smartMeter =
         this.domainHelperService.findSmartMeter(messageMetadata.getDeviceIdentification());
 
+    final Device connectToDevice;
+    if (smartMeter.getGatewayDevice() != null) {
+      connectToDevice = smartMeter.getGatewayDevice();
+    } else {
+      connectToDevice = smartMeter;
+    }
```

Comment:
Is this still necessary? We don't use GenerateAndReplaceKeys anymore for the installation of a G meter. Instead we use `SetEncryptionKeyExchangeOnGMeter`. Maybe add an errormessage to indicate that `GenerateAndReplaceKeys` is not supported for mbus meters?

Reply:
I removed this functionality